### DISCUSSION TITLE
Importing npm packages inside snap.tsx causes error

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1257] [Q&A]
While creating the snap if I import some npm package the build fails, also same happens while importing custom hooks i made outside the snaps directory, any way to bypass this ?

**Comment by ziad-saab** [comment]
@Arch0125 can you share some code please?

**Reply by Arch0125** [reply]
<img width="919" alt="Screenshot 2023-03-07 at 4 59 19 PM" src="https://user-images.githubusercontent.com/72722967/223409865-6e1edca1-1d41-4ab9-a3c7-2cef6239bf19.png">
If i try to do anything similar to this the build breaks, any work around ?

**Reply by Arch0125** [reply]
Not only ethers but any other npm package used inside index.tsx

**Reply by Arch0125** [reply]
Hey nvm got the solution here https://github.com/MetaMask/snaps-monorepo/blob/main/packages/examples/examples/ethers-js/src/index.js

**Reply by Arch0125** [reply]
I guess same could be done with biconomy sdk as well ?

**Reply by Montoya** [reply]
Yes the same should work with the Biconomy SDK. But you will need to check and see if the Biconomy SDK will run under Secure EcmaScript. If it helps, you can check this example snap that uses the SDK: https://github.com/livingrockrises/snap-auto-approvals

**Reply by Arch0125** [reply]
Okay that will help a lot, also i run into this frequently while importing packages is there something im doing wrong ?
<img width="515" alt="Screenshot 2023-03-07 at 10 48 51 PM" src="https://user-images.githubusercontent.com/72722967/223499308-11302c2c-70ee-45a8-923b-1d006549855d.png">
